### PR TITLE
runtime: Add test for epoch boundary

### PR DESCRIPTION
**Pull request by vadorovsky**
Add test that ensures correct values in `Bank` and `PartitionedRewardsCalculation` after crossing epoch boundary.

### CODE REVIEW

File: runtime/src/bank/partitioned_epoch_rewards/calculation.rs [hunk 1]
```diff
@@ -1523,18 +1517,16 @@ mod tests {
...
-            12392,                        // expected_rewards
-            8_400_000_000_000u128,        // expected_points
-            epoch_rewards_sysvar_balance, // capitalization_offset
-            None,                         // parent_capitalization
+            1,                     // expected_cache_len
+            &voters,               // expected_voters
+            &stakers,              // expected_stakers
+            0,                     // expected_vote_rewards
+            12300,                 // expected_stake_rewards
+            0.07599999499005672,   // expected_validator_rate
```
Comment (alexpyattaev) [inline runtime/src/bank/partitioned_epoch_rewards/calculation.rs]: This will break again, floating point is not stable across architectures and CPU releases...

Comment (vadorovsky) [reply runtime/src/bank/partitioned_epoch_rewards/calculation.rs]: good point, removed

**Review by alexpyattaev — DISMISSED** [review]
LGTM!

**Comment by vadorovsky** [comment]
~Putting this on hold, I'll submit a full relaxed test here once the revert gets merged.~ Ready for review.

**Comment by alexpyattaev** [comment]
The asserts look good, for the rest of the review I defer to the SMEs.

**Review by HaoranYi — DISMISSED** [review]

**Review by HaoranYi — APPROVED** [review]
